### PR TITLE
[ENH] Fill direction

### PR DESCRIPTION
**Pull request by samukweku**
# PR Description

Please describe the changes proposed in the pull request:

- Uses kwargs
- Limit parameter deprecated
-

**This PR improves fill_direction.**

# PR Checklist

Please ensure that you have done the following:

1. [x] PR in from a fork off your branch. Do not PR from `<your_username>`:`dev`, but rather from `<your_username>`:`<feature-branch_name>`.

2. [x] If you're not on the contributors list, add yourself to `AUTHORS.rst`.

3. [x] Add a line to `CHANGELOG.md` under the latest version header (i.e. the one that is "on deck") describing the contribution.
    - Do use some discretion here; if there are multiple PRs that are related, keep them in a single line.

# Automatic checks

There will be automatic checks run on the PR. These include:

- Building a preview of the docs on Netlify
- Automatically linting the code
- Making sure the code is documented
- Making sure that all tests are passed
- Making sure that code coverage doesn't go down.

# Relevant Reviewers

Please tag maintainers to review.

- @ericmjl

### CODE REVIEW

File: janitor/functions.py [hunk 1]
```diff
@@ -5300,68 +5287,75 @@ def fill_direction(
...
+    DOWNUP = "ffill", "bfill"
+
+
+def _chain_func(column: pd.Series, *funcs):
+    """
+    Apply series of functions consecutively
+    to a function.
+    https://blog.finxter.com/how-to-chain-multiple-function-calls-in-python/
+    """
+    new_value = column
```
Comment (nvamsikrishna05) [inline janitor/functions.py]: If inplace modification is being done to the column, then this would still modify the main `column`.

**Comment by nvamsikrishna05** [comment]
@samukweku Looks good.

File: janitor/functions.py [hunk 2]
```diff
@@ -2523,7 +2521,7 @@ def row_to_names(
         df = df.drop(df.index[range(row_number)])
 
     if reset_index:
-        df.reset_index(drop=["index"], inplace=True)
+        df = df.reset_index(drop=["index"])
```
Comment (ericmjl) [inline janitor/functions.py:2524]: :tada: I like this change!

File: janitor/functions.py [hunk 3]
```diff
@@ -37,6 +37,8 @@
 from scipy.stats import mode
 
 from .errors import JanitorError
+from enum import Enum
+from operator import methodcaller
```
Comment (ericmjl) [inline janitor/functions.py:41]: Oh my, that is such Python trickery, @samukweku :smile: 🤣

**Review by ericmjl — APPROVED** [review]
Lots of awesome ideas in here, approving, @samukweku :smile:.

**Comment by ericmjl** [comment]
@nvamsikrishna05 would you like to do the honors? Please hit the big green button at your convenience!

**Pull request merged**